### DISCUSSION TITLE
ContextManager.stopSpan

### DISCUSSION THREAD

**Discussion by kingkong2019** [apache/skywalking#8537] [Q&A]
When I need to invoke ContextManager.stopSpan ? I check many plugins code , but get no rules about it. 

I want to know if I put ContextManager.stopSpan in every afterMethod what will happen? Because I suspect if I dont do so there will be a span leak happened.

**Comment by wu-sheng** [comment]
Span has its own logic concept. Have you read the span concept in the Google Dapper Paper? I think that is a good start. We didn't set up a rule for it, because we simply can't. 
Whether `stopSpan` in `afterMethod` is based on your scenario, no standard answer.
